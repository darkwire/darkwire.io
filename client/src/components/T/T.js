import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getTranslations } from 'i18n';
import _ from 'lodash';

const regex = /{(.*?)}/g;

class T extends Component {
  render() {
    const t = getTranslations(this.props.language);
    const englishT = getTranslations('en');
    const str =
      _.get(t, this.props.path, '') || _.get(englishT, this.props.path, '');
    let string = str.split(regex);
    if (this.props.data) {
      string = string.map((word) => {
        if (this.props.data[word]) {
          return this.props.data[word];
        }
        return word;
      });
      return <span>{string}</span>;
    }
    return string;
  }
}

T.propTypes = {
  path: PropTypes.string.isRequired,
};

export default T;
