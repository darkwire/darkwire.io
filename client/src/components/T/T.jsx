import PropTypes from 'prop-types';
import _ from 'lodash';

import { getTranslations } from '@/i18n';

const regex = /{(.*?)}/g;

const T = ({ language, path, data }) => {
  const t = getTranslations(language);
  const englishT = getTranslations('en');
  const str = _.get(t, path, '') || _.get(englishT, path, '');

  let string = str.split(regex);

  // Data for string interpolation
  if (data) {
    string = string.map(word => {
      if (data[word]) {
        return data[word];
      }
      return word;
    });
    return <span>{string}</span>;
  }
  return string;
};

T.propTypes = {
  path: PropTypes.string.isRequired,
};

export default T;
