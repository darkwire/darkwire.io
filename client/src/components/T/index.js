import React, { Component } from 'react'
import PropTypes from 'prop-types'
import {getTranslations} from 'i18n';
import _ from 'lodash';
import { connect } from 'react-redux'

const regex = /{(.*?)}/g;

class T extends Component {
  render() {
    const t = getTranslations(this.props.language);
    let string = _.get(t, this.props.path, '').split(regex);
    if (this.props.data) {
      string = string.map(word => {
        if (this.props.data[word]) {
          return this.props.data[word];
        }
        return word;
      });
      return <span>{string}</span>
    }
    return string;
  }

  t() {
    const t = getTranslations(this.props.language);
    return _.get(t, this.props.path);
  }
}

T.propTypes = {
  path: PropTypes.string.isRequired,
}

export default connect(
  (state, ownProps) => ({
    language: state.app.language,
  }),
)(T)