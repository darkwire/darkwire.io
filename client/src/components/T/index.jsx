import { connect } from 'react-redux';
import T from './T';

export default connect((state, ownProps) => ({
  language: state.app.language,
}))(T);
