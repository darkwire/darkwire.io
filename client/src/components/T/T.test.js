import React from 'react';
import { render } from '@testing-library/react';

import T from './T';

// To avoid missing provider
jest.mock('components/T');

test('T component is displaying', async () => {
  const { asFragment, rerender } = render(<T path="welcomeHeader" language="en" />);

  expect(asFragment()).toMatchSnapshot();

  rerender(<T path="welcomeHeader" language="fr" />);

  expect(asFragment()).toMatchSnapshot();

  rerender(<T path="welcomeHeader" language="xx" />);

  expect(asFragment()).toMatchSnapshot();

  rerender(<T path="missingKey" language="en" />);

  expect(asFragment()).toMatchSnapshot();

  rerender(<T path="userJoined" language="en" data={{ username: 'Alan' }} />);

  expect(asFragment()).toMatchSnapshot();

  rerender(<T path="userJoined" language="en" />);

  expect(asFragment()).toMatchSnapshot();
});
