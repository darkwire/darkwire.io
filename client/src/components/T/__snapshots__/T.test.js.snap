// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`T component is displaying 1`] = `
<DocumentFragment>
  Welcome to Darkwire v2.0
</DocumentFragment>
`;

exports[`T component is displaying 2`] = `
<DocumentFragment>
  Bienvenue sur Darkwire v2.0
</DocumentFragment>
`;

exports[`T component is displaying 3`] = `
<DocumentFragment>
  Welcome to Darkwire v2.0
</DocumentFragment>
`;

exports[`T component is displaying 4`] = `<DocumentFragment />`;

exports[`T component is displaying 5`] = `
<DocumentFragment>
  <span>
    Alan joined
  </span>
</DocumentFragment>
`;

exports[`T component is displaying 6`] = `
<DocumentFragment>
  username joined
</DocumentFragment>
`;
